src/libraries/System.Private.CoreLib/src/System/Threading/Wasi/WasiPollWorld.wit.imports.wasi.io.v0_2_0.PollInterop.cs

```diff
@@ -0,0 +1,56 @@
+// Generated by `wit-bindgen` 0.27.0. DO NOT EDIT!
+// <auto-generated />
+#nullable enable
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Collections;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace WasiPollWorld.wit.imports.wasi.io.v0_2_0
+{
+    internal static class PollInterop {
+
+        internal static class PollWasmInterop
+        {
+            [DllImport("wasi:io/poll@0.2.0", EntryPoint = "poll"), WasmImportLinkage]
+            internal static extern void wasmImportPoll(nint p0, int p1, nint p2);
+
+        }
+
+        internal  static unsafe uint[] Poll(List<global::WasiPollWorld.wit.imports.wasi.io.v0_2_0.IPoll.Pollable> @in)
+        {
+
+            byte[] buffer = new byte[4 * @in.Count];
+            var gcHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
+            var address = gcHandle.AddrOfPinnedObject();
+
+            for (int index = 0; index < @in.Count; ++index) {
+                global::WasiPollWorld.wit.imports.wasi.io.v0_2_0.IPoll.Pollable element = @in[index];
+                int basePtr = (int)address + (index * 4);
+                var handle = element.Handle;
+                BitConverter.TryWriteBytes(new Span<byte>((void*)(basePtr + 0), 4), unchecked((int)handle));
```

Comment:
Under what conditions would `TryWriteBytes` fail?  The [docs](https://learn.microsoft.com/en-us/dotnet/api/system.bitconverter.trywritebytes) don't make that clear.  If it only fails when the `Span` is an incorrect size, then failure wouldn't be possible here, correct?

Reply:
maybe this should be just `Unsafe.Write<T>`

Reply:
There should be no little to no measurable difference between `BitConverter.TryWriteBytes` and respective `Unsafe.*` usage, expect using `Unsafe` or `MemoryMarshal` here would go against the efforts to reduce their use in the runtime.

Reply:
> expect using Unsafe or MemoryMarshal here would go against the ongoing efforts to reduce their use in the runtime.

This is using unmanaged pointers that are about as unsafe as `Unsafe....` or `MemoryMarshal`. Wrapping unmanaged pointer in a Span to right before dereferencing makes the code more complicated, but it does not make it safer.

Reply:
> This is using unmanaged pointers that are about as unsafe as `Unsafe....` or `MemoryMarshal`. Wrapping unmanaged pointer in a Span to right before dereferencing makes the code more complicated, but it does not make it safer.

That's true, but I think it would be general goodness if the wit-binder could get rid of the entire unmanaged pointer by `stackalloc`ating the Span (and removing the pinning) above and then using that Span directly. Potentially more complicating for the code generation, but (in my opinion atleast) much more safer modern .NET approach?